src/commands/init.ts

```diff
@@ -52,183 +53,138 @@ export default class Init extends Command {
-
-    if (!dirEmpty && (flags.force || (await cli.confirm('Directory is not empty. Should the project be initialized anyway?')))) {
-      params.push('--force');
-    }
-    await execa('npx', params, { cwd: flags.projectDir });
-
-    if (!fs.existsSync('.git')) {
-      await execa('git', ['init'], { cwd: flags.projectDir });
+  private async buildScaffold(projectDir: string, { force, verbose }: Flags) {
+    if (fs.readdirSync(projectDir).length !== 0) {
+      const dirString = projectDir === '.' ? 'this directory' : projectDir;
+      if (force || (await cli.confirm(`Directory is not empty. Remove all files in ${dirString}?`))) {
+        rm.sync(`${projectDir}/{*,.*}`);
+      }
     }
-    cli.action.stop();
-  }
 
-  private async getOptions(flags: Flags) {
+    const cliPath = path.resolve('node_modules/.bin/nest');
     const options: { [key: string]: string } = {
-      projectName:
-        flags.projectName ||
-        (await cli.prompt('Enter project name (for use in manifest.yml)', {
-          default: this.packageJson(flags).name
-        })),
-      command:
-        flags.startCommand ||
-        (await cli.prompt('Enter the command to start your server', {
-          default: this.packageJson(flags).scripts.start ? 'npm start' : ''
-        }))
+      cwd: projectDir,
+      stdio: verbose ? 'inherit' : 'ignore'
     };
 
-    return options;
-  }
-
-  private packageJson(flags: Flags) {
-    try {
-      if (fs.existsSync(path.resolve(flags.projectDir, 'package.json'))) {
-        return JSON.parse(
-          fs.readFileSync(path.resolve(flags.projectDir, 'package.json'), {
-            encoding: 'utf8'
-          })
-        );
-      }
-    } catch (error) {
-      this.error('Your package.json does not contain valid JSON. Please repair or delete it.', { exit: 10 });
-    }
-  }
-
-  private async modifyPackageJson(flags: Flags, initializationType: InitType) {
-    const packageJsonData = packageJsonParts(initializationType);
-    const { scripts, dependencies, devDependencies } = this.packageJson(flags);
-    const addFrontendScripts: boolean =
-      flags.frontendScripts ||
-      (!flags.skipFrontendScripts && (await cli.confirm('Should frontend-related npm scriptsToBeAdded for CI/CD be added?')));
-    const backendScritps = { ...packageJsonData.backendBuildScripts, ...packageJsonData.backendTestScripts };
-    const scriptsToBeAdded = addFrontendScripts ? { ...backendScritps, ...packageJsonData.frontendScripts } : backendScritps;
-
-    const conflicts = scripts ? Object.keys(scriptsToBeAdded).filter(name => Object.keys(scripts).includes(name)) : [];
-
-    if (
-      conflicts.length &&
-      !(await cli.confirm(`Script(s) with the name(s) "${conflicts.join('", "')}" already exist(s). Should they be overwritten?`))
-    ) {
-      this.error('Script exits as npm scripts could not be written.', {
-        exit: 11
-      });
+    if (fs.existsSync(cliPath)) {
```

Comment:
"If a full specifier is included, or if --package is used, npx will always use a freshly-installed, temporary version of the package." 🙁

Reply:
It's a "damned if you do, damned if you don't" situation. using full specifier always downloads, while just the command name leads to errors if the package is not already installed... https://github.com/zkat/npx/issues/151